### PR TITLE
Aspnet core windsor final expose cross wiring

### PR DESCRIPTION
**Pull request by generik0**
@fir3pho3nixx 
I have done a minimum of refactoring here.
But i believe this is the correct way to do this.

I know the add wire is exposed to the service collection, but I did a test, and seriously, you have to do some serious stuff to misuse it (especially out of the Startup.cs). We could change to internal if you really believe, but i think it should be in a separate file.

I need to add tests. This PR is for comments.
After any comments and tests added i believe we are ready to merge all the way to castle windsor @home.

I will make seperate PR for refactoring after we are done here.

### CODE REVIEW

**Comment by fir3pho3nixx** [comment]
Can we please close #14 first.

File: src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs [hunk 1]
```diff
@@ -335,7 +310,9 @@ private static void AddApplicationComponentsToWindsor(IWindsorContainer containe
 		private static void AddFrameworkResolversToWindsor(IServiceCollection services, IWindsorContainer container)
 		{
 			container.Kernel.Resolver.AddSubResolver(new LoggerDependencyResolver(services));
-			container.Kernel.Resolver.AddSubResolver(new FrameworkDependencyResolver(services));
+			var frameworkDependencyResolver = new FrameworkDependencyResolver(services);
```
[the remark this answers is not in the document]
Comment (generik0) [reply src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs]: I added it from your spike. It was needed to solve the issue.

File: src/Castle.Facilities.AspNetCore/Interfaces/IResetServiceProvider.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNetCore.Interfaces
+{
+	public interface IResetServiceProvider
```
[the remark this answers is not in the document]
Comment (generik0) [reply src/Castle.Facilities.AspNetCore/Interfaces/IResetServiceProvider.cs]: Same as above. It was to solve the issue at hand.

**Comment by generik0** [comment]
@fir3pho3nixx as mentioned above. I don’t previously I don’t see the difference between 14 and 15 as there are intertwined. And added on desperate commits, with id numbers.

I am getting to the point where I am committing the same thing over and over again.
Maybe it is just better you do it how you prefer. I think you know the goal, and the way to do it.
I still feel we should just throw an exception if late add (ie ServiceProvider not null)

I will cancel this PR. If you want me to review it when you make it, let me know.
Either way, I am happy we got to the bottom of this:-)

**Comment by fir3pho3nixx** [comment]
Pitty you feel that way. I was hoping we could have a chat tonight. I am not blocking you, I am trying to get you to engage me so you can explain why you feel these changes are important. I think you are a great developer, I just don’t understand why. Your failure to explain this or to even engage me in a chat is the last chance I am giving this. This is not me rejecting your changes. This is you failing to explain why...

**Comment by fir3pho3nixx** [comment]
Cancelling our goggle chat session tonight was not a good move either ...

**Comment by generik0** [comment]
@fir3pho3nixx, yes we should have chatted. But seriously, I had family objections.
This git issue dialogue can be difficult, and is not optial in any way. We seam to loose the context of the dialogue fast.
I see you update the PR to castle. Great work!
It has been fun looking at this issue with you, I got to understand castle core and windsor in a deeper level. And I also am very happy to get a facility that does the work we need...

I will let you know how it goes, if anything else pops up or if it is smooth sailing.
(By the way, I am still loosing sleep over the commit where I didn’t run tests. Crazy amateur mistake I should have known better).
If we need to do more work, I think it is a good idea you started with her, making issues. Doing one thing at a time is best :-)

Have a good day mate :-)

**Pull request closed without merging**